pkg/tfgen/generate_python.go

```diff
@@ -794,6 +793,82 @@ func (g *pythonGenerator) recordPropertyRec(sch *schema.Schema, info *tfbridge.S
 	}
 }
 
+// emitMembers emits property declarations and docstrings for all output properties of the given resource.
+func (g *pythonGenerator) emitMembers(w *tools.GenWriter, mod *module, res *resourceType) {
+	for _, prop := range res.outprops {
+		name := pyName(prop.name)
+		ty := pyType(prop)
+		w.Writefmtln("    %s: pulumi.Output[%s]", name, ty)
+		if prop.doc != "" {
+			g.emitDocComment(w, prop.doc, "    ")
+		}
+	}
+}
+
+// emitInitDocstring emits the docstring for the __init__ method of the given resource type.
+//
+// Sphinx (the documentation generator that we use to generate Python docs) does not draw a distinction between
+// documentation comments on the class itself and documentation comments on the __init__ method of a class. The docs
+// repo instructs Sphinx to concatenate the two together, which means that we don't need to emit docstrings on the class
+// at all as long as the __init__ docstring is good enough.
```

Comment:
One note - with this new approach VS Code no longer renders the help text for the class when hovering over the class name.

It does render the concatenation of class and constructor help text as parameter help on the constructor call.

Should we split these up and just let Sphinx re-concatenate them?  Seems that would result in good experience for both, and a generally more readable documentation in the code itself?

Reply:
We *could* actually put the docstring in both locations and then tell sphinx to ignore the class docstring - I think that would work.

Reply:
I actually sort of like keeping the class doc comment and the constructor doc comment separate.  Feels like there's a pretty natural way to do that (and it's the same we do for Node.js).  And it seems both Sphinx and VSCode can handle this well.  Any reason not to do that?

Reply:
I'll have to see what Sphinx does with `__init__` parameters - I'll try it today.

Reply:
@lukehoban I tried this today and Sphinx did not handle this very well - it re-rendered the class docstring as expected but it mangled the `__init__` docstring pretty badly.

I think I'd rather merge what we've got now and iterate on the dev experience as we go, is that OK?